agent_app_sample_code/00_global_config.py

```diff
@@ -0,0 +1,200 @@
+MLFLOW_EXPERIMENT_NAME = f"/Users/{user_email}/{AGENT_NAME}"
+
+# MLflow Run Names
+# These Runs will store your initial POC.  They are later used to evaluate the POC model against your experiments to improve quality.
+
+# Data pipeline MLflow run name
+POC_DATA_PIPELINE_RUN_NAME = "data_pipeline_poc"
+# Chain MLflow run name
+POC_CHAIN_RUN_NAME = "agent_poc"
+
+# COMMAND ----------
+
+print('--user info--')
+print(f"user_name {user_name}")
+
+print("--agent--")
+print(f"AGENT_NAME {AGENT_NAME}")
+print(f"UC_CATALOG {UC_CATALOG}")
+print(f"UC_SCHEMA {UC_SCHEMA}")
+print(f"UC_MODEL_NAME {UC_MODEL_NAME}")
+
+print()
+print("--evaluation config--")
+print(f"EVALUATION_SET_FQN {EVALUATION_SET_FQN}")
+print(f"MLFLOW_EXPERIMENT_NAME {MLFLOW_EXPERIMENT_NAME}")
+print(f"POC_DATA_PIPELINE_RUN_NAME {POC_DATA_PIPELINE_RUN_NAME}")
+print(f"POC_CHAIN_RUN_NAME {POC_CHAIN_RUN_NAME}")
+
+# COMMAND ----------
+
+# # Write the parts of the configuration that are needed by the Agent model to a YAML file
+# # This is used to securely provision these credentials to the deployed Agent
+
+# agent_resources = {
+#   "api_version": "1",
+#   "databricks": {
+#     "vector_search_index": [
+#       {
+#         "name": VECTOR_INDEX_NAME
+#       }
+#     ],
+#     "serving_endpoint": [
+#       {
+#         "name": LLM_ENDPOINT
+#       },
+#       {
+#         "name": EMBEDDING_MODEL_ENDPOINT
+#       }
+#     ]
+#   }
+# }
+
+# import yaml
+
+# # Specify the file path for the YAML file
+# file_path = 'agent_resources.yaml'
+
+# # Write the dictionary to a YAML file
+# with open(file_path, 'w') as file:
+#     yaml.dump(agent_resources, file)
+
+# COMMAND ----------
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.vectorsearch import EndpointStatusState, EndpointType
+from mlflow.utils import databricks_utils as du
+from databricks.sdk.service.serving import EndpointCoreConfigInput, EndpointStateReady
+from databricks.sdk.errors import ResourceDoesNotExist
+import os
+from databricks.sdk.service.compute import DataSecurityMode
+from pyspark.sql import SparkSession
+
+w = WorkspaceClient()
+browser_url = du.get_browser_hostname()
+
+# COMMAND ----------
+
+# Get the cluster ID
+spark_session = SparkSession.getActiveSession()
+cluster_id = spark_session.conf.get("spark.databricks.clusterUsageTags.clusterId", None)
+
+# # Get the current cluster name
+# try:
+#   cluster_id = dbutils.notebook.entry_point.getDbutils().notebook().getContext().tags().get("clusterId").get()
+# except Exception as e:
+
+cluster_info = w.clusters.get(cluster_id)
+
+# Check if a single user cluster
+# Serverless will return None here
+# if not cluster_info.data_security_mode == DataSecurityMode.SINGLE_USER:
+#   raise ValueError(f"FAIL: Current cluster is not a Single User cluster.  This notebooks currently require a single user cluster.  Please create a single user cluster: https://docs.databricks.com/en/compute/configure.html#single-node-or-multi-node-compute")
```

Comment:
Is this true? this might be a blocker for some customers.

Reply:
not true anymore. i confirmed its working on serverless and shared